### PR TITLE
Level expressions

### PR DESCRIPTION
**Pull request by jonsterling**
Depends on #387. This leads to a lot of simplification; it also makes it possible to pattern match in tactic scripts on level expressions.

Related to #301

### CODE REVIEW

**Comment by jonsterling** [comment]
This also fixes some bad bugs that I found along the way.

Still needed is to restore some of the fancier notation for level expressions.

**Comment by jonsterling** [comment]
I want to do the same thing to dimensions, by the way.

**Comment by jonsterling** [comment]
It will also be useful to do this to *kinds*, so that we can match on these in tactics.

**Comment by favonia** [comment]
I wonder what the bad bugs are along the way?

File: src/redprl/redprl.grm [hunk 1]
```diff
@@ -236,13 +236,21 @@ struct
 
   fun makeUseLemma (opid, params, subtermArgs) tacs = 
     let
-      val args = List.map (fn tac => ([],[]) \ tac) tacs
+      val args = subtermArgs @ List.map (fn tac => ([],[]) \ tac) tacs
```
Comment (jonsterling) [inline src/redprl/redprl.grm]: @favonia This is one of the "bad bugs"

Comment (jonsterling) [reply src/redprl/redprl.grm]: In fact, I can't remember, but this might be the only one.

**Pull request merged**